### PR TITLE
Lt 110 refactor create merch route

### LINKED ISSUE #110: feat:f1(4)refactor-create-merch-route

### PR DESCRIPTION
**Pull request by LeeTato**
# Changes 

-   add AUTH Handler and role Handler for create-merch 
-   add save() inside create-user
-  add role inside app-routing
# Purpose
 To protect the create-march route from the basic user
# Learning
N/A
Closes #110

### CODE REVIEW

File: src/client/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -16,7 +17,11 @@ const routes: Routes = [
   {path: 'create-message', component:ChatComponent},
   {path:'contact-page', component:ContactPageComponent},
   {path: 'create-merch', component: CreateMerchComponent},
-  {path: 'merch', component: MerchComponent},
```
Comment (brycejjacks) [inline src/client/src/app/app-routing.module.ts:19]: woah this looks like it still needs to be here.

File: src/client/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -16,7 +17,11 @@ const routes: Routes = [
...
+
+  {path: 'create-merch', component: CreateMerchComponent,
+  canActivate: [RolesGuard], data:{roles:["ADMIN"]}},
+
+
```
Comment (brycejjacks) [inline src/client/src/app/app-routing.module.ts]: I dont think we should adjust the style too much by adding all these empty lines

File: src/server/routers/user-route.ts [hunk 3]
```diff
@@ -28,6 +28,7 @@ userRouter.post("/create-user", async function (req:any, res:any) {
          roles:[role?._id]
          
         });
+        user.save().then(()=>res.status(200).json({data:user}))
     });
```
Comment (brycejjacks) [inline src/server/routers/user-route.ts:33]:
```suggestion
    .catch(err => res.status(501).json(err))
    });
```
i like having the catch there .. but not madatory of course.

**Review by vinsonf — APPROVED** [review]
Nice atomic commits! Thank you @LeeTato

**Pull request merged**